packages/medusa/src/services/claim.ts

```diff
@@ -204,6 +209,124 @@ export default class ClaimService extends TransactionBaseService {
     )
   }
 
+  protected async validateCreateClaimInput(
+    data: CreateClaimInput
+  ): Promise<void> {
+    const lineItemServiceTx = this.lineItemService_.withTransaction(
+      this.manager_
+    )
+
+    const { type, claim_items, additional_items, refund_amount } = data
+
+    if (type !== ClaimType.REFUND && type !== ClaimType.REPLACE) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claim type must be one of "refund" or "replace".`
+      )
+    }
+
+    if (type === ClaimType.REPLACE && !additional_items?.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claims with type "replace" must have at least one additional item.`
+      )
+    }
+
+    if (!claim_items?.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claims must have at least one claim item.`
+      )
+    }
+
+    if (refund_amount && type !== ClaimType.REFUND) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Claim has type "${type}" but must be type "refund" to have a refund_amount.`
+      )
+    }
+
+    const claimLineItems = await lineItemServiceTx.list(
+      { id: claim_items.map((c) => c.item_id) },
+      { relations: ["order", "swap", "claim_order", "tax_lines"] }
+    )
+
+    for (const line of claimLineItems) {
+      if (
+        line.order?.canceled_at ||
+        line.swap?.canceled_at ||
+        line.claim_order?.canceled_at
+      ) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Cannot create a claim on a canceled item.`
+        )
+      }
+    }
+  }
+
+  /**
+   * Finds claim line items on an order and calculates the refund amount.
+   * There are three places too look:
+   * - Order items
+   * - Swap items
+   * - Claim items (from previous claims)
+   * Note, it will attempt to return early from each of these places to avoid having to iterate over all items every time.
+   * @param order - the order to find claim lines on
+   * @param claimItems - the claim items to match against
+   * @return the refund amount
+   */
+  protected async getRefundTotalForClaimLinesOnOrder(
+    order: Order,
+    claimItems: CreateClaimItemInput[]
+  ) {
+    const claimLines = claimItems
+      .map((ci) => {
+        const predicate = (it: LineItem) =>
+          it.shipped_quantity! > 0 &&
+          ci.quantity <= it.shipped_quantity! &&
+          it.id === ci.item_id
```

Comment:
self-review: This is the updated check. The claim item quantity should be less than or equal to the amount actually shipped. And if that holds, there's no need to check the fulfilled quantity.